zsl/interface/celery/worker.py

```diff
@@ -0,0 +1,108 @@
+"""
+:mod:`zsl.interface.celery.worker`
+----------------------------------
+
+Implementation of celery workers.
+
+.. moduleauthor:: Peter Morihladko
+"""
+from __future__ import unicode_literals
+
+import sys
+
+from celery import Celery, shared_task
+from injector import Module, singleton, provides
+
+from zsl import Config
+from zsl.utils.injection_helper import inject
+from zsl.interface.task_queue import TaskQueueWorker
+from zsl.task.job_context import Job
+
+
+class CeleryTaskQueueWorkerBase(TaskQueueWorker):
+    """Base class for celery task queue worker.
+
+    It initialize the celery app in constructor.
+    """
+
+    @inject(config=Config)
+    def __init__(self, config):
+        # type: (Config) -> None
+        super(CeleryTaskQueueWorkerBase, self).__init__()
+
+        self.celery_app = Celery()
+        self.celery_app.config_from_object(config['CELERY'])
+
+    def execute_celery_task(self, job_data):
+        # type: (dict) -> dict
+        """Creates job from task and executes the job.
+
+        :param job_data: job's data
+        :return: job results
+        :rtype: dict
+        """
+        job = Job(job_data)
+
+        return self.execute_job(job)
+
+
+class CeleryTaskQueueOutsideWorker(CeleryTaskQueueWorkerBase):
```

Comment:
This is my answer to a problem, where we have the worker as executioner of task queue requests and the worker it self. What happens when only the task execution is needed and the worker is manage outside with some other tool (celeryd)? I just puts some error logs to the "worker control" functions.